src/Enigma2.cpp

```diff
@@ -1774,51 +1800,41 @@ PVR_ERROR Enigma2::GetDriveSpace(long long *iTotal, long long *iUsed)
     XMLUtils::GetString(hddNode, "e2capacity", capacity);
     XMLUtils::GetString(hddNode, "e2free", freeSpace);
 
-    std::regex regexGB ("^.* GB");
-    std::regex regexMB ("^.* MB");
-    std::regex regexReplaceGB (" GB");
-    std::regex regexReplaceMB (" MB");
-    std::string replaceWith = "";
+    totalKb += GetKbFromString(capacity);
+    freeKb += GetKbFromString(freeSpace);
+  }
 
-    if (regex_match(capacity, regexGB))
-    {
-      //It's GB
-      double gigabytesSize = atof(regex_replace(capacity, regexReplaceGB, replaceWith).c_str());
+  *iTotal = totalKb;
+  *iUsed = totalKb - freeKb;
 
-      total += (long long)(gigabytesSize * 1024 * 1024);
-    }
-    else
-    {
-      //It's MB
-      if (regex_match(capacity, regexMB))
-      {
-        double megabytesSize = atof(regex_replace(capacity, regexReplaceMB, replaceWith).c_str());
+  Logger::Log(LEVEL_INFO, "GetDriveSpace Total: %u, Used %u", *iTotal, *iUsed);
```

Comment:
Are you sure printf'ing a `long long`usinf `%u` works? Isn't it `%lld` or that PRId64 stuff from inttypes.h ?

Reply:
Done